src/nacl/hash.py

```diff
@@ -17,10 +17,80 @@
 import nacl.bindings
 import nacl.encoding
 
+BYTES = nacl.bindings.crypto_generichash_BYTES
+BYTES_MIN = nacl.bindings.crypto_generichash_BYTES_MIN
+BYTES_MAX = nacl.bindings.crypto_generichash_BYTES_MAX
+KEYBYTES = nacl.bindings.crypto_generichash_KEYBYTES
+KEYBYTES_MIN = nacl.bindings.crypto_generichash_KEYBYTES_MIN
+KEYBYTES_MAX = nacl.bindings.crypto_generichash_KEYBYTES_MAX
+SALTBYTES = nacl.bindings.crypto_generichash_SALTBYTES
+PERSONALBYTES = nacl.bindings.crypto_generichash_PERSONALBYTES
+
+_b2b_hash = nacl.bindings.crypto_generichash_blake2b_salt_personal
+
 
 def sha256(message, encoder=nacl.encoding.HexEncoder):
     return encoder.encode(nacl.bindings.crypto_hash_sha256(message))
 
 
 def sha512(message, encoder=nacl.encoding.HexEncoder):
     return encoder.encode(nacl.bindings.crypto_hash_sha512(message))
+
+
+def generichash(data, digest_size=BYTES, key=b'',
```

Comment:
I'm dubious about whether such a generic-sounding name as `generichash` should be exposed. I know nacl likes to offer a short(er) name for the specific recommended algorithm for each class of operations, but I don't think hiding the specific hash being used serves anybody very well. Maybe we should leave this function out, and just offer `blake2b()` instead?